src/core/sync/mutex.d

```diff
@@ -163,7 +167,7 @@ class Mutex :
         lock_nothrow();
     }
 
-    // undocumented function for internal use
+    /// ditto
     final void lock_nothrow(this Q)() nothrow @trusted @nogc
```

Comment:
Are you documenting these because the normal versions can't be made `nothrow` and `@nogc` without breaking code? You _can_ overload on attributes, but that still breaks code since you get an error about overriding one of the functions shadowing the others. I can see someone overriding `lock` and `unlock` for debugging purposes, but it seems really dumb for the version of the function that you would ideally call to not be `lock` or `unlock`... Also, the `_nothrow` function names don't actually follow the official naming convention, though `lockNothrow` does seem a bit ugly. As such, I'd be inclined to find a better name for them, but I don't know what it would be. Something like `lockMutex` would follow the naming convention and doesn't seem as ugly, but it seems redundant, and it doesn't make the difference between it and `lock` obvious.